creator/ingest_runs/common/model.py

```diff
@@ -0,0 +1,182 @@
+import hashlib
+import uuid
+import re
+
+import django_rq
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db import models
+from django_fsm import FSMField, transition
+
+from creator.jobs.models import JobLog
+from creator.utils import stop_job
+
+User = get_user_model()
+
+
+def camel_to_snake(camel_str):
+    """
+    Convert CamelCase to snake_case
+    """
+    return re.sub(r'(?<!^)(?=[A-Z])', '_', camel_str).lower()
+
+
+def hash_versions(versions):
+    """
+    Uniquely identify a set of file Versions by computing an MD5 hash digest
+    using the primary keys of the Versions
+    """
+    version_id_str = "".join(sorted(v.kf_id for v in versions))
+    return hashlib.md5(version_id_str.encode("utf-8")).hexdigest()
+
+
+class State(object):
+    NOT_STARTED = "not_started"
+    RUNNING = "running"
+    FAILED = "failed"
+    CANCELED = "canceled"
+    COMPLETED = "completed"
+
+
+class IngestProcess(models.Model):
+    """
+    Common model functionality for ingest processes
+    (e.g. ingest run, validation run)
+    """
+    __queue__ = "ingest"
+
+    class Meta:
+        abstract = True
+
+    id = models.UUIDField(
+        primary_key=True, default=uuid.uuid4, editable=False
+    )
+    created_at = models.DateTimeField(
+        auto_now_add=True,
+        null=True,
+        help_text="Time when the ingest process was created",
+    )
+    creator = models.ForeignKey(
+        User,
+        null=True,
+        blank=True,
+        related_name="%(class)ss",
```

Comment:
I'm a bit confused here, why no `'%(app_label)s'`?

Reply:
I'm also confused. Why are these old string formats being used instead of `ingest_processes` or something similar?

Reply:
Django's model inheritance rules say that you must specify a unique name for `related_name` so we can't just make it `ingest_processes` (and also I think that would be a little confusing since we have two different kinds of ingest processes). 

They do recommend using both the app label and the class name to guarantee uniqueness (see https://docs.djangoproject.com/en/3.1/topics/db/models/#be-careful-with-related-name-and-related-query-name) but I thought that this `User.ingest_runs_ingestruns` looks gross vs `User.ingestruns` which is more aligned with how we specify `related_name` on all our other models.

Reply:
I see. To get closer to what we use on other fields, maybe you want to use that handy `camel_to_snake()` here to turn it into `ingest_runs`?
Also, I think you have one too many `s` here.

Reply:
Lol that's what I thought too with that extra `s`, but `%(class)s` gets replaced with `ingestrun` so you need the extra `s` for it to be `ingestruns`

Reply:
I'm not sure how easy it is or if its possible to set `related_name` with a function..I'd have to look into it